grpclb/src/main/java/io/grpc/grpclb/GrpclbState.java

```diff
@@ -252,6 +269,7 @@ void handleAddresses(
       }
       // Start the fallback timer if it's never started
       if (fallbackTimer == null) {
+        fallbackReason = BALANCER_TIMEOUT_STATUS;
```

Comment:
Put this in FallbackModeTask? That'd also be more obviously before maybeUseFallbackBackends(). I recognize it'd have to be passed to the constructor.

Reply:
Ah, that was what my previous revision look like. But I was afraid of missing cases that just let the fallback timer fire and reply on condition checks in `maybeUseFallbackBackends()` to skip (re-) entering fallback. It looks all cases that certainly not entering fallback or entering fallback early are cancelling the timer. So this should be fine.

Reply:
It turned out that I did miss one case: we need to short-circuit the fallback timer when the balancer RPC closes, instead of just let it execute. Although fallback logic won't be executed again as it's already in fallback, it will overwrite the fallback reason. To avoid this happen, we need to ensure the fallback timer is cancelled for all cases that we enter fallback early.